shared/core/types/relay.go

```diff
@@ -0,0 +1,53 @@
+package types
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+)
+
+var (
+	errInvalidRelayInvalidPayload  = errors.New("invalid relay payload")
+	errInvalidJSONRPCInvalidRPC    = errors.New("invalid value for JSONRPC field")
+	errInvalidJSONRPCMissingMethod = errors.New("Method field not set")
+	errInvalidRESTPayload          = errors.New("invalid REST payload")
+)
+
+// INCOMPLETE: perform any possible metadata validation
+// Validate performs validation on the relay payload
+func (r Relay) Validate() error {
+	if jsonRpcPayload := r.GetJsonRpcPayload(); jsonRpcPayload != nil {
+		return jsonRpcPayload.Validate()
+	}
+
+	if jsonPayload := r.GetRestPayload(); jsonPayload != nil {
+		return jsonPayload.Validate()
+	}
+
+	return fmt.Errorf("%w: %v", errInvalidRelayInvalidPayload, r)
+}
+
+// Validate performs validation on JSONRPC payload. More specifically, it verifies that:
+//  1. The JSONRPC field is set to "2.0" as per the JSONRPC spec requirement, and
+//  2. The Method field is not empty
+func (p JSONRPCPayload) Validate() error {
+	if p.JsonRpc != "2.0" {
```

Comment:
Consider moving "2.0" into a local constant

Reply:
Thank you, fixed.